pandas/core/arrays/period.py

```diff
@@ -85,94 +108,113 @@ def wrapper(self, other):
+    - freq : pd.tseries.offsets.Tick
 
-    @property
-    def freq(self):
-        """Return the frequency object if it is set, otherwise None"""
-        return self._freq
-
-    @freq.setter
-    def freq(self, value):
-        msg = ('Setting {cls}.freq has been deprecated and will be '
-               'removed in a future version; use {cls}.asfreq instead. '
-               'The {cls}.freq setter is not guaranteed to work.')
-        warnings.warn(msg.format(cls=type(self).__name__),
-                      FutureWarning, stacklevel=2)
-        self._freq = value
+    The values are physically stored as a 1-D ndarray of integers. These are
+    called "ordinals" and represent some kind of offset from a base.
 
-    # --------------------------------------------------------------------
-    # Constructors
+    The `freq` indicates the span covered by each element of the array.
+    All elements in the PeriodArray have the same `freq`.
 
+    See Also
+    --------
+    period_array : Create a new PeriodArray
+    pandas.PeriodIndex : Immutable Index for period data
+    """
     _attributes = ["freq"]
+    _typ = "periodarray"  # ABCPeriodArray
+
+    # Names others delegate to us
+    _other_ops = []
+    _bool_ops = ['is_leap_year']
+    _object_ops = ['start_time', 'end_time', 'freq']
+    _field_ops = ['year', 'month', 'day', 'hour', 'minute', 'second',
+                  'weekofyear', 'weekday', 'week', 'dayofweek',
+                  'dayofyear', 'quarter', 'qyear',
+                  'days_in_month', 'daysinmonth']
+    _datetimelike_ops = _field_ops + _object_ops + _bool_ops
+    _datetimelike_methods = ['strftime', 'to_timestamp', 'asfreq']
 
-    def __new__(cls, values, freq=None, **kwargs):
-        if is_period_dtype(values):
-            # PeriodArray, PeriodIndex
-            if freq is not None and values.freq != freq:
-                raise IncompatibleFrequency(freq, values.freq)
-            freq = values.freq
-            values = values.asi8
-
-        elif is_datetime64_dtype(values):
-            # TODO: what if it has tz?
-            values = dt64arr_to_periodarr(values, freq)
-
-        return cls._simple_new(values, freq=freq, **kwargs)
+    # --------------------------------------------------------------------
+    # Constructors
+    def __init__(self, values, freq=None, copy=False):
+        # type: (Union[PeriodArray, np.ndarray], Union[str, Tick]) -> None
+        if isinstance(values, ABCSeries):
+            values = values._values
+            if not isinstance(values, type(self)):
+                raise TypeError("Incorrect dtype")
+
+        elif isinstance(values, ABCPeriodIndex):
+            values = values._values
+
+        if isinstance(values, type(self)):
+            if freq is not None and freq != values.freq:
+                raise TypeError("freq does not match")
+            values, freq = values._data, values.freq
+
+        values = np.array(values, dtype='int64', copy=copy)
```

Comment:
I think we *always* need to copy here, e.g. if you pass in an ndarray with copy=False this makes the internals mutable which is not great

Reply:
I think we should certainly have the *option* to not copy. 
Assume things like `take` or `fillna`: here we already create a new ndarray, that does not need to be copied. 

Can you explain a bit more in detail what you are thinking about, where you see a problem in the internals?

Reply:
IMO, this should be like `np.array`: a function that takes some stuff and turns it into an array (`PeriodArray` in this case)`.

The default there is `copy=True`, so let's follow that/

Reply:
(deleted an incorrect comment).

I think I was following Series, Index, and IntegerArray, IntervalArray, SparseArray, which don't copy (index is special because it's immutable).

We should be internally consistent among the array classes. `Categorical` doesn't have a copy parameter.

Reply:
this is very tricky. If there are outside refernces here the I think we *need* to copy. You are right technically we don't need to and for Series we do have this behavior. But this is an outside leakage. I guess ``copy=True`` as a default just works here then.